src/adaptors/tcp_adaptor.c

```diff
@@ -502,13 +502,27 @@ static void handle_disconnected(qdr_tcp_connection_t* conn)
         qd_log(tcp_adaptor->log_source, QD_LOG_DEBUG,
                "[C%"PRIu64"][L%"PRIu64"] handle_disconnected - close out_dlv_stream",
                conn->conn_id, conn->outgoing_id);
+
+        // Fun fact: the delivery pointed to by conn->initial_delivery is
+        // eventually moved to conn->out_dlv_stream.  I don't trust the code
+        // enough to be confident both pointers are not set at the same time
+        // and we risk calling remote_state_updated twice.
+        if (conn->initial_delivery == 0) {
```

Comment:
Because of the fact that `conn->initial_delivery` was set to zero in a couple of places elsewhere in the code, the need to check `conn->initial_delivery == 0` arises here. From what I can tell, there is no need to set `conn->initial_delivery` to zero since we are never reusing non-egress dispatcher connection objects. If we don't ever set the `conn->initial_delivery` to zero, then there will be no need to check if `conn->initial_delivery == 0`

Reply:
Yeah but that's like... out of scope for this patch IMHO. Not a bad idea, but I'd hold off that change until after the release.